### DISCUSSION TITLE
What makes your bot profitable

### DISCUSSION THREAD

**Discussion by yuichiroaoki** [yuichiroaoki/poly-flashloan-bot#77] [Ideas]
## Reasons why your bot is profitable
80%: High market volatility

15%:  Execution speed

5%: Target trading routes/pairs

### Setting Gas Price Dynamic
You can get the current gas price with ethers.js.   
You should add some value to the current gas price for your tx to get included faster.  

ex.
```typescript
  const gasPrice = await provider.getGasPrice();
  const extraGas = ethers.utils.parseUnits("100", "gwei");

  return Flashloan.connect(signer).dodoFlashLoan(params, {
    gasLimit: gasLimit,
    gasPrice: gasPrice.add(extraGas),
  });
```
Note: This way your bot will consume a lot of Matic when the market is volatile and gas price are high. 

### Running a Polygon Node
You can get price faster and there's no rate limits.  
cost: $470 USD / month (using GCP) 

Ref. https://docs.polygon.technology/docs/integrate/full-node-deployment/

**Comment by recidive** [comment]
Hello @yuichiroaoki !

Is there anything besides increasing gas prices that increases the chance the transaction gets into the next block?

No matter how much gas I add my transactions take 2-3 blocks to get confirmed.

I posted a question [here](https://github.com/yuichiroaoki/poly-flash/discussions/62), but it looks like this repo is more active.

Thank you very much!
